### PR TITLE
Full Stack Development

### PR DESCRIPTION
**Pull request by stevenweber**
This builds off the commits in #1, with some noticeable changes:

* The UI is completely isolated, no artifacts are built into the API
* `bin/foreman start` boots API and UI for single-point development
* Leveraging Rack Builders to create, most importantly, a single mount point for integration-level testing

### CODE REVIEW

File: features/support/env.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+if headless
+  Capybara.register_driver(:poltergeist) do |app|
+    Capybara::Poltergeist::Driver.new(app, { debug: false })
+  end
+else
+  Capybara.register_driver(:browser) do |app|
+    Capybara::Selenium::Driver.new(app, { browser: :chrome })
+  end
+end
+Capybara.default_driver = Capybara.javascript_driver = (headless ? :poltergeist : :browser)
```
Comment (gmassanek) [inline features/support/env.rb]: Cool. I didn't love forcing poltergeist/phantom.

File: lib/builders.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require "rack"
```
Comment (gmassanek) [inline lib/builders.rb]: Interesting approach. Hadn't considered Rack::Static.

File: lib/tasks/ui.rake [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require "fileutils"
+
+namespace(:ui) do
+  task(:build) do
+    original_dir = FileUtils.pwd
+
+    begin
+      FileUtils.cd("agile-direction-ui")
+      system("npm install")
```
Comment (gmassanek) [inline lib/tasks/ui.rake]: Does bower install happen automagically?

**Comment by gmassanek** [comment]
Is public/assets/.keep required anymore? Doesn't look like anything uses it.

File: lib/tasks/ui.rake (hunk 3 above)
Comment (stevenweber) [inline lib/tasks/ui.rake]: It's [part of the postinstall](https://github.com/agile-direction/agile-direction-ui/blob/master/package.json#L14).

**Pull request merged**